### PR TITLE
fix(discover) Add withProjects() to fix missing projects

### PR DESCRIPTION
**Pull request by markstory**
I missed a spot where we were using `organization.projects` in discover results. This fixes that and ensures projects are loaded in performance views in case we start using projects there too.

Fixes JAVASCRIPT-21Z0

### CODE REVIEW

**Review by billyvg — APPROVED** [review]
This looks good and will address the bug, but I think what we might want to do is have the fieldRenderer use `app/utils/projects` to fetch the project details in the renderer (it uses the ProjectsStore as cache).

I don't think we should continue to assume `withProjects` will return the list of all projects.

**Comment by billyvg** [comment]
Looking at this further, it seems a bit cleaner to move the project logic into the field renderer itself instead of requiring the tables to use the HoC

**Comment by markstory** [comment]
@billyvg I'll give that a shot.

File: src/sentry/static/sentry/app/utils/discover/fieldRenderers.tsx [hunk 1]
```diff
@@ -198,10 +199,18 @@ const SPECIAL_FIELDS: SpecialFields = {
...
         <Container>
-          {project ? <ProjectBadge project={project} avatarSize={16} /> : data.project}
+          <Projects orgId={organization.slug} slugs={[data.project]}>
+            {({projects}) => {
+              const project = projects.find(p => p.slug === data.project);
+              return project ? (
+                <ProjectBadge project={project} avatarSize={16} />
+              ) : (
+                data.project
+              );
```
Comment (billyvg) [inline src/sentry/static/sentry/app/utils/discover/fieldRenderers.tsx:211]: Should we always render project badge? 

```suggestion
              return <ProjectBadge project={project ? project : {slug: data.project}} avatarSize={16} />
```

Aside, there's a prop to passthrough slug as a placeholder: https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/utils/projects.tsx#L85 so this should always find a project and have a placeholder. (Which I think would make a better experience).

**Review by billyvg — APPROVED** [review]
lgtm

**Pull request merged**